geomesa-hbase/geomesa-hbase-rpc/src/main/scala/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.scala

```diff
@@ -0,0 +1,82 @@
+/***********************************************************************
+ * Copyright (c) 2013-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.hbase.filters
+
+import org.apache.hadoop.hbase.exceptions.DeserializationException
+import org.geotools.filter.text.ecql.ECQL
+import org.locationtech.geomesa.hbase.rpc.filter.CqlTransformFilter
+import org.locationtech.geomesa.hbase.rpc.filter.CqlTransformFilter.{DelegateFilter, FilterDelegate, FilterTransformDelegate, NullFeatureIndex, TransformDelegate, deserialize}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.apache.hadoop.hbase.util.Bytes
+import org.locationtech.geomesa.features.SerializationOption.SerializationOptions
+import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer
+
+@deprecated("used for filter compatibility mode 2.3")
+class JSimpleFeatureFilter(delegate: DelegateFilter) extends CqlTransformFilter(delegate) {
```

Comment:
Why does this file need to move?  (Just double-checking/curious.)

Reply:
it's used transparently to the user when they configure the CqlTransformFilter, which is in the rpc module, so i had to move it there so it could be seen: https://github.com/locationtech/geomesa/pull/2693/files/48fd3e44fec88de2698398ca02d76dacafebf2d1#diff-84beca7a6920b5da6a09b1c8deebae38990a1ec7f19ff8d80039feb502067e7aR124

Reply:
i think it was an oversight that it wasn't moved there before